### ISSUE TITLE
deck show with -s flag

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#71]
I've tested the `deck show` command with the `-s` flag:
First of all I've listed the PoD token decks using `deck list -p` command and got the following output
```
| Local label      | ID                                                               | Global name               | Issuer                             | M | Conf.  | I |
+------------------+------------------------------------------------------------------+---------------------------+------------------------------------+---+--------+---+
| DTNewSpecv2_newl | a2459e054ce0f600c90be458915af6bad36a6863a0ce0e33ab76086b514f765a | DTNewSpecv2               | mvrm2HAoKqiCmeQiwKMuEFpeEn7rJEmpMz | 1 | 270179 | + |
|                  | 3369f3eb703656c0b46b613dcfb4df78fc05657a0b75f44551634de6321d9d9c | dt_short                  | mwzox84gMy8TKdeBftrfpmwGVZnJJdaReg | 1 | 263638 | + |
|                  | 0b3816628374d24078c741251f712b7f6af1c6f82ff7ddce083c20a58fcc2bc1 | impossible_donation_token | mpmZCVusQ6WoWHcFTAizmEcqSXdu5YX5PV | 1 | 31669  |   |
```

Then I've picked ` DTNewSpecv2_newl` deck and run `deck show  DTNewSpecv2_newl  -s`. The output was:

```
        General error raised by PeerAssets.

        Check if your input is correct.

        If you gave a deck as an argument, a possible reason for this error is that you need to initialize the deck.

        To initialize the default decks, use:

        pacli deck init

        To initialize a single deck, use:

        pacli deck init DECKID
```
Then 
- I've re-initialized the deck using `deck init a2459e054ce0f600c90be458915af6bad36a6863a0ce0e33ab76086b514f765a` command,
- tried `deck show a2459e054ce0f600c90be458915af6bad36a6863a0ce0e33ab76086b514f765a  -s`, `deck show DTNewSpecv2  -s`,
- restarted `slimcoind` with and without `-rescan` flag.
But got always the same output.

**Comment by d5000** [comment]
Fixed the error because it should not appear in this case and confuses the user.

But you can't user `deck show -s` without `-i` (the `-s` will then be ignored). This is also mentioned in the docstring.

**Comment by buhtignew** [comment]
This works.

> This is also mentioned in the docstring.

Maybe we could stress it even more editing `Shows P2TH addresses (in combination with -i/--info, only dPoD tokens)` into `Shows P2TH addresses (only in combination with -i/--info and only for dPoD token decks)`?

**Comment by d5000** [comment]
Done, will be uploaded in the next relevant commit. Can be closed if there are no further questions.

**Comment by buhtignew** [comment]
I'm reopening this issue because it seems like the `-s` flag provides additional info not only for the dPoD tokens decks but for all the others as well.
Here is the output of the `token show ATTokenNewSpec1 -i -s` command on my side (in case of PoB token):
```    'version': 1,
    'name': 'ATTokenNewSpec2',
    'issue_mode': 1,
    'number_of_decimals': 4,
    'asset_specific_data': b'\x10\x02@dJ\x14@\xf1c\xa4\xd0\xa8\xbcD\xb4\xba\x00\xb9T\xc2\xbd\xbe\xfb\x87|\xf4P\x02',
    'id': 'fb93cce7aceb9f7fda228bc0c0c2eca8c56c09c1d846a04bd6a59cae2a895974',
    'issuer': 'mvrm2HAoKqiCmeQiwKMuEFpeEn7rJEmpMz',
    'issue_time': 1679503968,
    'tx_confirmations': 314244,
    'network': 'tslm',
    'production': True,
    'at_address': 'mmSLiMCoinTestnetBurnAddress1XU5fu',
    'multiplier': 100,
    'addr_type': 2,
    'startblock': None,
    'endblock': None,
    'at_type': 2,
    '_p2th_address': 'mkwJijAwqXNFcEBxuC93j3Kni43wzXVuik',
    'derived_p2th_addresses': {
        'proposal': 'mzFbLbvssaPMwzocVyezn6C6WeUL8Skjxj',
        'signalling': 'mv5Rb3RYDBqVHhPRFzbN1nUuNsBySWLJf3',
        'locking': 'mwQUHrpkBrhLiRDiRshDdf1mBQtecvdX81',
        'donation': 'mugJcj4zJ2X8ZVqkGwYEBvyeZNZBTubRad',
        'voting': 'mtx7h8hsFmHGVNEQ7P5eEfNTfcbwLGCKB3'
```
And here is the output of the `token show ATTokenNewSpec1 -i -s` command I've got (the case of ATToken):
```
    'version': 1,
    'name': 'ATTokenNewSpec',
    'issue_mode': 1,
    'number_of_decimals': 2,
    'asset_specific_data': b'\x10\x02@\x01J\x14e|\x17\x93\xfe+\xbf\xeaj%r\xbdZ\x00\xdcJ\x1e\xf0\x07\xb0P\x02',
    'id': '7a2ae406ddf44ddb17532d4888fdc14573f52749445e9e014075c9f83cbe556f',
    'issuer': 'mpmZCVusQ6WoWHcFTAizmEcqSXdu5YX5PV',
    'issue_time': 1709053770,
    'tx_confirmations': 76196,
    'network': 'tslm',
    'production': True,
    'at_address': 'mpmZCVusQ6WoWHcFTAizmEcqSXdu5YX5PV',
    'multiplier': 1,
    'addr_type': 2,
    'startblock': None,
    'endblock': None,
    'at_type': 2,
    '_p2th_address': 'mz6HENDiSFPYqsi7wDZ33p5jt7j9LFo6qr',
    'derived_p2th_addresses': {
        'proposal': 'mk5K8d4cA89m44V9dkPCssHK4pWqnRSyQp',
        'signalling': 'miuT6cGZ6bz2VzjmF1oVgzk6wM35fAdu4o',
        'locking': 'mqBCYo2KpKK9W7axKwdKNHWdLAAzV1iLyx',
        'donation': 'mvjmJCyFVokr6BsDkT5G393kVhhxzHMgJA',
        'voting': 'mwVpXuzixKixoDChUiuDbzLqpJgz9WJFsh'
  ```
So I was wondering whether it's the right information we are providing in the `-s` flag description of the `token show` command: `Shows P2TH addresses (only in combination with -i/--info, only dPoD tokens)`, maybe the part `, only dPoD tokens` is not exact?

**Comment by d5000** [comment]
Fixed in commit 7a0e42f, the proposal/signalling/locking etc. address of course should not be shown for PoB tokens, because they will never be needed (they're generated by a deterministic algorithm, so in theory you can derive them from each deck ID).

I later saw that the help was also incorrect as you pointed out. The main P2TH address will always be shown if `-s` is added, regardless of it's a dPoD token or not. This will be commited with the next update.

**Comment by buhtignew** [comment]
This is solved that's why I'm closing it.
